### PR TITLE
Bit90 - New driver for Bit90 computer by Bit Corporation

### PR DESCRIPTION
**Pull request by palazzol**
Actually, 2 drivers:
Working driver is bit90b3 - BASIC 3.0 EPROMS from older unit
bit90 will be BASIC 3.1 Mask ROMs, currently documented but undumped

### CODE REVIEW

File: src/mame/drivers/coleco.cpp [hunk 1]
```diff
@@ -91,6 +96,40 @@ WRITE8_MEMBER( coleco_state::paddle_on_w )
...
+	//osd_printf_verbose("Bankswitch to u4\n");
+	m_bank->set_entry(0);
+	return 0xff;
+}
+
+READ8_MEMBER( bit90_state::bankswitch_u3_r )
+{
+	//osd_printf_verbose("Bankswitch to u3\n");
+	m_bank->set_entry(1);
+	return 0xff;
```
Comment (cuavas) [inline src/mame/drivers/coleco.cpp]: Three things here:
* You should return `space.unmap()` for the unmapped reads
* You should guard against debugger access in read handlers with side effects by checking `if (!machine().side_effects_disabled())`
* For logging that can be disabled at compile time, you should use the `LOG` macro from `logmacro.h`.

File: src/mame/drivers/coleco.cpp [hunk 2]
```diff
@@ -91,6 +96,40 @@ WRITE8_MEMBER( coleco_state::paddle_on_w )
...
+WRITE8_MEMBER( bit90_state::u32_w )
+{
+	// Write to a 74LS174 at u32
+	// Bits 4-7 are connected for keyboard scanning (actually only 4-6 are used)
+	// Bits 0-1 unknown (probably tape?)
+	m_keyselect = (data >> 4) & 0x0f;
+	uint8_t temp = data & 0x03;
+	if (m_unknown != temp) {
+		m_unknown = temp;
+		osd_printf_verbose("m_unknown -> 0x%02x\n",m_unknown);
```
Comment (cuavas) [inline src/mame/drivers/coleco.cpp]: This should use `logerror`

File: src/mame/drivers/coleco.cpp [hunk 3]
```diff
@@ -485,6 +686,21 @@ ROM_END
...
+	ROM_LOAD("bit90b3.u2", 0x2000, 0x2000, CRC(b992b940) SHA1(c7dd96a1944fac40cbae20630f303a69de7e6313))
+ROM_END
+
+ROM_START( bit90 )
+	ROM_REGION( 0x4000, "banked", 0 )
+	ROM_LOAD("d32351e.u4", 0x0000, 0x2000, NO_DUMP) // BIT-99C1
+	ROM_LOAD("d32521e.u3", 0x2000, 0x2000, NO_DUMP) // MONITOR2
+	ROM_REGION( 0x10000, "maincpu", 0 )
+	ROM_LOAD("d32522e.u2", 0x2000, 0x2000, NO_DUMP) // MONITOR3
+ROM_END
```
Comment (cuavas) [inline src/mame/drivers/coleco.cpp]: Are these actually different machines, or different ROM revisions for the same machine?  If they're ROM revisions for the same machine, they should use the BIOS mechanism for selection.

**Comment by palazzol** [comment]
@cuavas Thanks for the excellent feedback, I will fix all these issues with another submission soon.  The two drivers are different ROM revisions of the same machine.

**Pull request merged**